### PR TITLE
Add ConstructorTile

### PR DESCRIPTION
**Pull request by Kroppeb**

### CODE REVIEW

File: src/main/kotlin/com/refinedmods/refinedstorage/extensions/Deprecated.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import net.minecraft.nbt.ListTag
+import net.minecraft.util.Hand
+import net.minecraft.util.Identifier
+import net.minecraft.util.hit.BlockHitResult
+import net.minecraft.util.math.Direction
+import net.minecraft.world.World
+import java.util.*
+
+@Deprecated("fabric", replaceWith = ReplaceWith("this.getStackInHand(hand)"))
+fun PlayerEntity.getHeldItem(hand: Hand): ItemStack = getStackInHand(hand)
```
Comment (thinkslynk) [inline src/main/kotlin/com/refinedmods/refinedstorage/extensions/Deprecated.kt:27]: This entire file is very smart. Me like.

File: src/main/kotlin/com/refinedmods/refinedstorage/util/WorldUtils.kt [hunk 2]
```diff
@@ -38,18 +40,21 @@ fun getFluidHandler(@Nullable tile: BlockEntity?, side: Direction?): IFluidHandl
...
+        /*
+        if (owner != null) {
+            val profileCache: PlayerProfileCache = world.getServer().getPlayerProfileCache()
+            val profile: GameProfile = profileCache.getProfileByUUID(owner)
+            if (profile != null) {
+                return FakePlayerFactory.get(world, profile)
+            }
         }
+        return FakePlayerFactory.getMinecraft(world)
+         */
```
Comment (thinkslynk) [inline src/main/kotlin/com/refinedmods/refinedstorage/util/WorldUtils.kt]: Can you undo the changes to this file so we don't get a conflict with my gui pr?

**Review by thinkslynk — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/refinedmods/refinedstorage/extensions/Deprecated.kt (hunk 1 above)
Comment (Kroppeb) [reply src/main/kotlin/com/refinedmods/refinedstorage/extensions/Deprecated.kt:26]: done
